src/main/java/frc/robot/subsystems/intake/Intake.java

```diff
@@ -0,0 +1,49 @@
+package frc.robot.subsystems.intake;
+
+import edu.wpi.first.units.Angle;
+import edu.wpi.first.units.MutableMeasure;
+import edu.wpi.first.units.Velocity;
+import edu.wpi.first.wpilibj2.command.Command;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.robot.subsystems.intake.IntakeConstants.IntakePose;
+import org.littletonrobotics.junction.Logger;
+
+public class Intake extends SubsystemBase {
+    private static Intake INSTANCE = null;
+    private final IntakeIO io;
+    private final IntakeInputsAutoLogged inputs = IntakeIO.inputs;
+
+    public Intake(IntakeIO io) {
+        this.io = io;
+    }
+
+    public Intake getInstance() {
+        return INSTANCE;
+    }
+
+    public void initialize(IntakeIO io) {
+        INSTANCE = new Intake(io);
+    }
+
+    public Command setAngle(MutableMeasure<Angle> angle) {
+        return runOnce(() -> io.setAngle(angle));
+    }
+
+    public Command setAngle(IntakePose intakePose) {
+        return setAngle(intakePose.intakePose);
+    }
+
+    public Command setRollerSpeed(MutableMeasure<Velocity<Angle>> speed) {
+        return runOnce(() -> io.setRollerSpeed(speed));
+    }
+
+    public Command setCenterRollerSpeed(MutableMeasure<Velocity<Angle>> speed) {
+        return runOnce(() -> io.setCenterRollerSpeed(speed));
+    }
+
+    @Override
+    public void periodic() {
+        io.updateInputs();
+        Logger.processInputs("Intake", inputs);
```

Comment:
Java provides a function that returns this class name: `this.getClass().getSimpleName()`. It should be cached (saved in a constant) and used here instead of a string -- this is useful if the class' name changes.

Reply:
ok, thanks

Reply:
8a21fe042ec15dac5f577d40cbb692c39c3aec95